qiskit_ibm_runtime/base_primitive.py

```diff
@@ -0,0 +1,37 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Base primitive"""
+
+from abc import ABC, abstractmethod
+from typing import Optional, Union, Any
+
+from .ibm_backend import IBMBackend
+from .ibm_runtime_service import IBMRuntimeService
+
+
+class BasePrimitive(ABC):
+    """Base primitive"""
+
+    def __init__(
+        self,
+        service: IBMRuntimeService,
+        backend: Optional[Union[IBMBackend, str]] = None,
+    ):
+        self._service = service
```

Comment:
Need to make `service` optional and if not passed try to instantiate `service = IBMRuntimeService()`. (If user has saved default credentials it'll work otherwise will throw error).

Reply:
Fixed.